packages/app-desktop/gui/MainScreen/commands/perNotebookSortOrder.ts

```diff
@@ -0,0 +1,212 @@
+		if (next.field != field || next.reverse != reverse) {
+			void CommandService.instance().execute(NOTES_SORT_ORDER_SWITCH, next.field, next.reverse);
+		}
+	}
+}
+
+const OWN_SORT_ORDERS = 'notes.perNotebookSortOrders';
+const SHARED_SORT_ORDER = 'notes.sharedSortOrder';
+const SUFFIX_FIELD = '$field';
+const SUFFIX_REVERSE = '$reverse';
+
+function getSelectedNotebookId(state?: State): string {
+	const s = state ? state : app().store().getState();
+	if (s.notesParentType === 'Folder') {
+		return s.selectedFolderId;
+	} else {
+		return '';
+	}
+}
+
+function getCurrentField(): string {
+	return Setting.value(SETTING_FIELD);
+}
+
+function getCurrentReverse(): boolean {
+	return Setting.value(SETTING_REVERSE);
+}
+
+function perFieldReversalEnabled(): boolean {
+	return Setting.value(SETTING_PER_FIELD_REVERSAL_ENABLED);
+}
+
+function perNotebookSortOrderEnabled(): boolean {
+	return Setting.value(SETTING_PER_NOTEBOOK_SORT_ORDER_ENABLED);
+}
+
+function getOwnSortOrder(notebookId: string) {
+	const field = ownSortOrders[notebookId + SUFFIX_FIELD] as string;
+	const reverse = ownSortOrders[notebookId + SUFFIX_REVERSE] as boolean;
+	return { field, reverse };
+}
+
+function setOwnSortOrder(notebookId: string, field: string, reverse: boolean) {
+	const old = getOwnSortOrder(notebookId);
+	let dirty = false;
+	if (old.field !== field) {
+		ownSortOrders[notebookId + SUFFIX_FIELD] = field;
+		dirty = true;
+	}
+	if (old.reverse !== reverse) {
+		ownSortOrders[notebookId + SUFFIX_REVERSE] = reverse;
+		dirty = true;
+	}
+	if (dirty) {
+		Setting.setValue(OWN_SORT_ORDERS, { ...ownSortOrders });
+	}
+}
+
+function deleteOwnSortOrder(notebookId: string) {
+	let dirty = false;
+	if (ownSortOrders.hasOwnProperty(notebookId + SUFFIX_FIELD)) {
+		delete ownSortOrders[notebookId + SUFFIX_FIELD];
+		dirty = true;
+	}
+	if (ownSortOrders.hasOwnProperty(notebookId + SUFFIX_REVERSE)) {
+		delete ownSortOrders[notebookId + SUFFIX_REVERSE];
+		dirty = true;
+	}
+	if (dirty) {
+		Setting.setValue(OWN_SORT_ORDERS, { ...ownSortOrders });
+	}
+}
+
+function loadOwnSortOrders() {
+	ownSortOrders = { ...Setting.value(OWN_SORT_ORDERS) };
+}
+
+function loadSharedSortOrder() {
+	const validFields = notesSortOrderFieldArray();
+	const value = Setting.value(SHARED_SORT_ORDER);
+	for (const key in sharedSortOrder) {
+		if (value.hasOwnProperty(key)) {
+			if (key !== 'field' || validFields.includes(value.field)) {
+				sharedSortOrder[key] = value[key];
+			}
+		}
+	}
+}
+
+function setSharedSortOrder(field: string, reverse: boolean) {
+	let dirty = false;
+	if (sharedSortOrder.field !== field) {
+		sharedSortOrder.field = field;
+		dirty = true;
+	}
+	if (sharedSortOrder.reverse !== reverse) {
+		sharedSortOrder.reverse = reverse;
+		dirty = true;
+	}
+	if (sharedSortOrder[field] !== reverse) {
+		sharedSortOrder[field] = reverse;
+		dirty = true;
+	}
+	if (dirty) {
+		Setting.setValue(SHARED_SORT_ORDER, { ...sharedSortOrder });
+	}
+}
```

Comment:
In fact, most of these functions should be moved to a utils package. Could you move them under maybe /services/perFolderSortOrder/utils.ts? Unless you can find a better place.

Reply:
Migrating to more appropriate venues is very fine.

One suggestion is that most part of perFolderSortOrder.ts and notesSortOrderSwitch.ts are moved to the followings respectively.
- app-desktop/services/perFolderSortOrderUtils.ts
- app-desktop/services/notesSortOrderUtils.ts

Reply:
I would prefer if those related functions are grouped so either in separate files but in a new folder, or both in the same file.

Reply:
I see. Final locations are:
- app-desktop/services/sortOrder/notesSortOrderUtils.ts
- app-desktop/services/sortOrder/PerFolderSortOrderService.ts

finished. 36e00f8